include/crow/websocket.h

```diff
@@ -94,6 +96,24 @@ namespace crow
                     return;
                 }
 
+                std::string requested_subprotocols_header = req.get_header_value("Sec-WebSocket-Protocol");
+                if (!subprotocols.empty() || !requested_subprotocols_header.empty())
+                {
+                    auto requested_subprotocols = utility::split(requested_subprotocols_header, ", ");
+                    auto subprotocol = utility::find_first_of(subprotocols.begin(), subprotocols.end(), requested_subprotocols.begin(), requested_subprotocols.end());
+                    if (subprotocol == subprotocols.end())
+                    {
+                        adaptor_.close();
+                        handler_->remove_websocket(this);
```

Comment:
I would avoid that as it's usually a client-side issue. Still, I remember a close code that could be related to that but I don't remember what it is...

Reply:
I am probably wrong, and this is not my area of expertise, and I do not feel strongly about this, but as I understand it, the client submits an ordered list of options and the server uses the first one it recognises.

It would seem seem that bad matches should be ignored and the connection left open, with no references to an option in the header.

Reply:
I will remove the closing part then.
Should I change the std::string to a std::unique_ptr<std::string> and return null in the "get_subprotocol()" method if no subprotocol matches ? Or do I keep it as an std::string and leave it empty when there's no match ?

Reply:
I'd prefer to avoid std::unique_ptr<> as:
- I'm generally keen to avoid things that hit a heap because of the implicit lock when used naively.
- The general style of the project is to just use strings, and in this case, and empty string is a good representation of "no match".

If we were using C++17, we could consider std::optional<std::string>.  However, we're C++11 compatible, and string will do.

Further more, common std::string implementations, have small string optimization, which means no allocation for small/empty strings.

So I'd say, use an empty string to represent no match.

Thanks for considering my input.

Reply:
Perfect, I made the modifications. Tell me if there's anything else

Reply:
I'm happy for the code to be merged.  You'll need to poke a project maintainer.